src/Application/Channels/Commands/CreateChannelCommand.cs

```diff
@@ -89,10 +81,37 @@ public async Task<Guid> Handle(CreateChannelCommand request, CancellationToken c
             entity.RangeRule = "*";
         }
 
+        entity.ActiveRevision = await GetActiveRevision(request, entity, cancellationToken);
+
         _context.Channels.Add(entity);
 
         await _context.SaveChangesAsync(cancellationToken);
 
         return entity.Id;
     }
+
+    private async Task<Revision?> GetActiveRevision(CreateChannelCommand request, Channel entity, CancellationToken cancellationToken)
+    {
+        if (entity.RevisionSelectionStrategy == ChannelRevisionSelectionStrategy.UseRangeRule &&
+                    request.ActiveRevisionId is null)
+        {
+            var revisions = _context.Revisions
+                .Where(c => c.AppId == request.AppId).ToList();
+
+            var revision = RevisionRangeRule.Parse(entity.RangeRule).Match(revisions);
+            return revision;
```

Comment:
Maybe declare a `Revision? revision = null;` at the top of the method and return only once at the end?